api/blueprints/drbCollection.py

```diff
@@ -63,28 +63,51 @@ def collectionCreate(user=None):
     collectionData = request.json
     dataKeys = collectionData.keys()
 
-    if len(set(dataKeys) & set(['title', 'creator', 'description'])) < 3\
-            or len(set(dataKeys) & set(['workUUIDs', 'editionIDs'])) == 0:
+    if len(set(dataKeys) & set(['title', 'creator', 'description'])) < 3:
         errMsg = {
-            'message':
-                'title, creator and description fields are required'
-                ', with one of workUUIDs or editionIDs to create a collection'
+            'message': 'title, creator and description fields are required',
         }
 
         return APIUtils.formatResponseObject(400, 'createCollection', errMsg)
 
     dbClient = DBClient(current_app.config['DB_CLIENT'])
     dbClient.createSession()
 
-    newCollection = dbClient.createCollection(
-        collectionData['title'],
-        collectionData['creator'],
-        collectionData['description'],
-        user,
-        workUUIDs=collectionData.get('workUUIDs', []),
-        editionIDs=collectionData.get('editionIDs', []),
-        type='static'
-    )
+    if "workUUIDs" in collectionData or "editionIDs" in collectionData:
+        newCollection = dbClient.createStaticCollection(
+            collectionData['title'],
+            collectionData['creator'],
+            collectionData['description'],
+            user,
+            workUUIDs=collectionData.get('workUUIDs', []),
+            editionIDs=collectionData.get('editionIDs', []),
+        )
+
+    elif "autoDef" in collectionData:
+        autoDef = collectionData["autoDef"]
+        queryFields = ["keywordQuery", "authorQuery", "titleQuery", "subjectQuery"]
+        errMsg = _validateAutoCollectionDef(autoDef)
+        if errMsg:
+            return APIUtils.formatResponseObject(400, "createCollection", errMsg)
+        newCollection = dbClient.createAutomaticCollection(
+            collectionData['title'],
+            collectionData['creator'],
+            collectionData['description'],
+            owner=user,
+            sortField=autoDef.get("sortField"),
+            sortDirection=autoDef.get("sortDirection"),
+            limit=autoDef.get("limit"),
+            **{field: autoDef.get(field) for field in queryFields},
+        )
```

Comment:
What would happen if a user puts in the editionIDs/workUUIDs and 'autodef' in the request json body? Would the ids be prioritized or the 'autodef' parameter. I think this edge case should be considered.

Reply:
Fair point! Right now, we'd just ignore the `autoDef` but it probably makes sense to raise an error if they do that, I can add that!

Reply:
Sounds good.